src/packet.cpp

```diff
@@ -20,6 +20,31 @@ Packet::Packet(const std::array<uint8_t, 4096>& m_recvBuffer, size_t bytes)
 
     for (std::size_t i = 0; i < m_auth.size(); ++i)
         m_auth[i] = m_recvBuffer[i + 4];
+
+    size_t attributeIndex = 20;
+    while (attributeIndex < length)
+    {
+        const uint8_t attributeType = m_recvBuffer[attributeIndex];
+        const uint8_t attributeLength = m_recvBuffer[attributeIndex + 1];
+
+        if (attributeType == 1)
+        {
+            m_attributes.push_back(new String(attributeType, &m_recvBuffer[attributeIndex + 2], attributeLength - 2));
+        }
+        else if (attributeType == 5)
```

Comment:
This chain of `if`s will grow quite big when you add more attribute- and data-types. And the sole purpose of these `if`s is to turn a triple (type, data pointer, data size) into an attribute pointer. Sounds like a definition of a function! So how about putting the `if`s into a function like this:
```c++
Attribute* makeAttribute(uint8_t type, const uint8_t* data, size_t size);

// ...

while (...)
{
    // ...
    try
    {
        m_attributes.push(makeAttribute(attributeType, &m_recvBuffer[attributIndex + 2], attributeLangth - 2));
    }
    catch (...)
    {
        // ...
    }
}
```